From 985a382e1e1f0b0ee2eb63ddcecd2800ee822a3e Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@nxp.com>
Date: Tue, 6 Mar 2018 12:46:18 +0800
Subject: [PATCH 3439/5242] MLK-17156-6: ASoC: imx-pcm-rpmsg: fix get codec
 data failed

commit  463ddb8b1b2af136114a4959f7b4e07a3a5949a0 from
https://source.codeaurora.org/external/imx/linux-imx.git

Receive message is only used when the type is B. originally
we copy the receive message to revg_msg all the time, when
the message type is C, which will overide the revg_msg, which
cause the get codec data command return wrong value.

Signed-off-by: Shengjiu Wang <shengjiu.wang@nxp.com>
Reviewed-by: Daniel Baluta <daniel.baluta@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/imx-pcm-rpmsg.c |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/sound/soc/fsl/imx-pcm-rpmsg.c b/sound/soc/fsl/imx-pcm-rpmsg.c
index 80a00a4..abfe8d2 100644
--- a/sound/soc/fsl/imx-pcm-rpmsg.c
+++ b/sound/soc/fsl/imx-pcm-rpmsg.c
@@ -467,8 +467,6 @@ static int i2s_rpmsg_cb(struct rpmsg_device *rpdev, void *data, int len,
 
 	dev_dbg(&rpdev->dev, "get from%d: cmd:%d.\n", src, msg->header.cmd);
 
-	memcpy(&i2s_info_g->recv_msg, msg, sizeof(struct i2s_rpmsg_r));
-
 	if (msg->header.type == I2S_TYPE_C) {
 		if (msg->header.cmd == I2S_TX_PERIOD_DONE) {
 			spin_lock_irqsave(&i2s_info_g->lock[0], flags);
@@ -487,8 +485,10 @@ static int i2s_rpmsg_cb(struct rpmsg_device *rpdev, void *data, int len,
 		}
 	}
 
-	if (msg->header.type == I2S_TYPE_B)
+	if (msg->header.type == I2S_TYPE_B) {
+		memcpy(&i2s_info_g->recv_msg, msg, sizeof(struct i2s_rpmsg_r));
 		complete(&i2s_info_g->cmd_complete);
+	}
 
 	return 0;
 }
-- 
1.7.9.5

